RedHat-UX/red-hat-design-system
# fix(tooltip): announce content to assistive technology

## What I did

1. Tooltip content is put in a `.visually-hidden` `<div>` with `aria-live` in the `<body>` so that it can be read by AT.
1. Added functionality where hitting ESC closes the `rh-tooltip`.
1. ~~Added an `aria-describedby`/`id` pair to elements in the lightdom of `<rh-tooltip>`~~
1. ~~Added Markup Guidance to the [Accessibility page](https://deploy-preview-2069--red-hat-design-system.netlify.app/elements/tooltip/accessibility) of `<rh-tooltip>` docs.~~

## Testing Instructions

1. Check out the [tooltip demo](https://deploy-preview-2069--red-hat-design-system.netlify.app/elements/tooltip/demo/) in the DP.
1. Test tooltip with various combinations of assistive tech
1. ~~Check out the Accessibility page, proofread the Markup Guidance, give feedback.~~

## Notes to Reviewers

~~I couldn't get `customElements.whenDefined()` working in place of `setTimeout()`. PTAL and maybe you will have better luck.~~

~~This PR changes the trigger `<rh-button>` elements in the demos to native `<button>` elements so that tooltip can be accessible.~~

Closes #2051.
